From 0658c73855dce856115a71015412450146ea6975 Mon Sep 17 00:00:00 2001
From: Daniel Wagner <daniel.wagner@bmw-carit.de>
Date: Wed, 5 Mar 2014 10:21:10 +0100
Subject: [PATCH 2/7] session: Add set_active_session() helper

With adding this helper function we are able to add a one single hook
to the policy plugin instead of several places.
---
 src/session.c | 21 +++++++++++----------
 1 file changed, 11 insertions(+), 10 deletions(-)

diff --git a/src/session.c b/src/session.c
index 8f9db49..7a786a8 100644
--- a/src/session.c
+++ b/src/session.c
@@ -416,6 +416,12 @@ static void free_session(struct connman_session *session)
 	g_free(session);
 }
 
+static void set_active_session(struct connman_session *session, bool enable)
+{
+	__connman_service_set_active_session(enable,
+				session->info->config.allowed_bearers);
+}
+
 static void cleanup_session(gpointer user_data)
 {
 	struct connman_session *session = user_data;
@@ -426,8 +432,7 @@ static void cleanup_session(gpointer user_data)
 	cleanup_firewall_session(session);
 
 	if (session->active)
-		__connman_service_set_active_session(false,
-				session->info->config.allowed_bearers);
+		set_active_session(session, false);
 
 	session_deactivate(session);
 	update_session_state(session);
@@ -897,8 +902,7 @@ int connman_session_config_update(struct connman_session *session)
 		&allowed_bearers);
 
 	if (session->active)
-		__connman_service_set_active_session(false,
-				session->info->config.allowed_bearers);
+		set_active_session(session, false);
 
 	session->active = false;
 	session_deactivate(session);
@@ -941,8 +945,7 @@ static DBusMessage *connect_session(DBusConnection *conn,
 
 	if (!session->active) {
 		session->active = true;
-		__connman_service_set_active_session(true,
-				session->info->config.allowed_bearers);
+		set_active_session(session, true);
 	}
 
 	session_activate(session);
@@ -968,8 +971,7 @@ static DBusMessage *disconnect_session(DBusConnection *conn,
 
 	if (session->active) {
 		session->active = false;
-		__connman_service_set_active_session(false,
-				session->info->config.allowed_bearers);
+		set_active_session(session, false);
 	}
 
 	session_deactivate(session);
@@ -1012,8 +1014,7 @@ static DBusMessage *change_session(DBusConnection *conn,
 				return __connman_error_failed(msg, -err);
 
 			if (session->active)
-				__connman_service_set_active_session(false,
-					session->info->config.allowed_bearers);
+				set_active_session(session, false);
 
 			session->active = false;
 			session_deactivate(session);
-- 
1.8.5.3

